### PR TITLE
Move contents of terraform repository

### PR DESCRIPTION
**Pull request by katieclaiborne-duet**
## [Airtable record](https://airtable.com/applRnjY08XXmLq3A/tbl9NslAfxuW3Etym/viwfIezxjk0NokBJ1/recSLm3OVpToRi6ty?blocks=hide)

Unblocks https://github.com/Duet-Technologies/terraform/pull/20

## Problem
We found ourselves with Terraform configurations in three GitHub repositories, and decided to [consolidate](https://docs.google.com/document/d/1U8428yRtgdSxmkC0ZLqTqn28ijMfkRHShpKee7LkiSg/edit).

## Solution
- Move the contents of the `terraform` GitHub repository into the `terraform` directory of this repository.
  - Remove GitHub resources from state (see note [here](https://docs.google.com/document/d/1U8428yRtgdSxmkC0ZLqTqn28ijMfkRHShpKee7LkiSg/edit#heading=h.vz7txy6er3n))
  - Import `data_exchange` resources into state

## Testing
- [x] `terraform plan` returns no changes

image

## Post-merge to-do
- [ ] Delete `terraform/data-exchange` prefixed state file from GCS

### CODE REVIEW

File: terraform/.terraform.lock.hcl [hunk 1]
```diff
@@ -21,21 +21,41 @@ provider "registry.terraform.io/hashicorp/archive" {
...
+    "zh:215f675174d4074f11226f5c48b75880b59c2d56f0ae22ae3ea16012023b68e3",
+    "zh:22bc73738a7a57e3a022b28bdb34039da07cc105c1f46b288002c02b05c05a71",
+    "zh:4876a02bc040c98fb662e985a78caecd2b6a7355480362648ff36826e45d09fa",
+    "zh:899a64484ea44092dd0822478032e87f661dd85e8d735adbb553f7796d7ee497",
+    "zh:991990de697bd32066e4f2b67aa10c59b84e18f75170be976e75fb711cd0070f",
+    "zh:a4db4d424c060c0e8f686b0fb714aa7d41e3a5ba20d6aba05496eb8c6e635a3d",
+    "zh:a988d8a24e673aeff6b1a298d731ed69ef63bba7af9cd4ceb5c51834eecf9fde",
+    "zh:be3855a320cc44b1936c0778c9cbcaebe92435a2d5eef1f6e5a967ce2a3dc01e",
+    "zh:e26c72bc15ae27a3a6a99ed0f60ed38f2fca54a310c7ddfee3126b93fb90509e",
+    "zh:efbe757f0764a1dd69d23a440c03eaf69a8e5742ad63e814212611a16d769eb6",
```
Comment (katieclaiborne-duet) [inline terraform/.terraform.lock.hcl:58]: See https://github.com/Duet-Technologies/terraform/pull/13/files#r1669127414.

File: terraform/main.tf [hunk 2]
```diff
@@ -1,11 +1,14 @@
 provider "google" {
   project = local.project_id
   region  = "us-central1"
+  
+  billing_project       = local.project_id
+  user_project_override = true
```
Comment (katieclaiborne-duet) [inline terraform/main.tf:6]: See https://github.com/Duet-Technologies/terraform/pull/16/files#r1681248040.

File: README.md [hunk 3]
```diff
@@ -1 +1 @@
-# data_exchange
+# data-infrastructure
```
Comment (katieclaiborne-duet) [inline README.md:1]: I renamed the repository manually, and hadn't yet updated the README header in code.

**Review by JoeVieira — APPROVED** [review]
looks good, basically a cut and paste to the new repo. nice work

**Comment by katieclaiborne-duet** [comment]
thanks, joe!

**Pull request merged**